mmv1/third_party/terraform/acctest/tgc_utils.go

```diff
@@ -1,64 +1,307 @@
+			}
+
+			// If ImportStateId is explicitly set, use that
+			if nextStep.ImportStateId != "" {
+				metadata.Id = nextStep.ImportStateId
+				return metadata
+			}
+
+			// If ImportStateIdPrefix is set, note it
+			if nextStep.ImportStateIdPrefix != "" {
+				metadata.Id = fmt.Sprintf("%s<AUTO_ID>", nextStep.ImportStateIdPrefix)
+				return metadata
 			}
-			caiAssetName := fmt.Sprintf("//%s/%s", apiServiceName, rName)
-			log.Printf("[DEBUG]TGC CAI asset names start\n%s\nEnd of TGC CAI asset names", caiAssetName)
+
+			// If ImportStateIdFunc is set, get function info
+			if nextStep.ImportStateIdFunc != nil {
+				metadata.Id = "<DYNAMIC_IMPORT_ID>"
+				return metadata
+			}
+
+			// Default case - the ID will be automatically determined
+			metadata.Id = "<AUTO_ID>"
+			return metadata
+		}
+	}
+
+	return metadata
+}
+
+// extendWithTGCData adds TGC metadata check function to the last non-plan config entry
+func extendWithTGCData(c resource.TestCase) resource.TestCase {
+	var updatedSteps []resource.TestStep
+
+	// Find the last non-plan config step
+	lastNonPlanConfigStep := -1
+	for i := len(c.Steps) - 1; i >= 0; i-- {
+		step := c.Steps[i]
+		if step.Config != "" && !step.PlanOnly {
+			lastNonPlanConfigStep = i
+			break
 		}
+	}
 
-		// The acceptance tests names will be also used for the tgc tests.
-		// "service" is logged and will be used to put the tgc tests into specific service packages.
-		log.Printf("[DEBUG]TGC Terraform service: %s", service)
-		log.Printf("[DEBUG]TGC Terraform resource: %s", address)
+	// Process all steps
+	for i, step := range c.Steps {
+		// If this is the last non-plan config step, add our TGC check
+		if i == lastNonPlanConfigStep {
+			// Parse resources from the config
+			resources := parseResources(step.Config)
 
-		re := regexp.MustCompile(`\"(tf[-_]?test[-_]?.*?)([a-z0-9]+)\"`)
-		rawConfig = re.ReplaceAllString(rawConfig, `"${1}tgc"`)
-		log.Printf("[DEBUG]TGC raw_config starts %sEnd of TGC raw_config", rawConfig)
-		return nil
+			// Skip if no resources found
+			if len(resources) == 0 {
+				updatedSteps = append(updatedSteps, step)
+				continue
+			}
+
+			// Determine the service package from the first resource
+			firstResource := resources[0]
+			parts := strings.Split(firstResource, ".")
+			if len(parts) < 2 {
+				updatedSteps = append(updatedSteps, step)
+				continue
+			}
+
+			// Collect metadata for all resources
+			resourceMetadata := make(map[string]ResourceMetadata)
+			for _, res := range resources {
+				parts := strings.Split(res, ".")
+				if len(parts) >= 2 {
+					resourceType := parts[0]
+
+					// Determine import metadata if the next step is an import step
+					importMeta := determineImportMetadata(c.Steps, i, res)
+
+					// Create metadata for this resource
+					resourceMetadata[res] = ResourceMetadata{
+						ResourceType:    resourceType,
+						ResourceAddress: res,
+						ImportMetadata:  importMeta,
```

Comment:
Is `Service` missing?

Reply:
Service was removed from the test and put on the resource directly

Reply:
we could use reflection on the testing object to find where which service the test was defined in but there are two things here. The service the resource is defined in and the service the test is defined in.

Reply:
ah sorry you are right. I added it